network/p2p/libp2pNode.go

```diff
@@ -797,8 +797,9 @@ func DefaultLibP2POptions(address string, key fcrypto.PrivateKey) ([]config.Opti
 	// as the 1-k discovery process and the 1-1 messaging both sometimes attempt to open connection to the same target
 	// As of now there is no requirement of client sockets to be a well-known port, so disabling port reuse all together.
 	transport := libp2p.Transport(func(u *tptu.Upgrader) *tcp.TcpTransport {
-		tpt := tcp.NewTCPTransport(u)
-		tpt.DisableReuseport = true
+		// TODO: handle errors properly. DisableReuseport() does not return an error, so it's safe
+		// to ignore for now
+		tpt, _ := tcp.NewTCPTransport(u, tcp.DisableReuseport())
```

Comment:
We'd better return the error. The desired case is to fail safely when we cannot bootstrap the node completely.